EIPS/eip-draft_request_id.md

```diff
@@ -0,0 +1,135 @@
+
+**Q: The efficency gains might encourage clients to flood their peers with too many simultanous requests**
+
+Peers can always throttle or disconnect if they don't feel treated well. This is the same as today.
+
+**Q: If `les` already defines the commands like this, why not just use the `les` protocol?**
+
+In practice, peers that serve the `les` protocol are much harder to find in the network. The reasons for this are varied but might boil down to client defaults, immature implementations or missing incentives.
+
+**Q: Networking works today, isn't this just adding bloat?**
+
+This is adding a single integer per command while at the same time reducing code complexity and improving networking effiency. The addition seems justified.
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+This EIP extends the `eth` protocol in a backwards incompatible way and requires rolling out a new version, `eth/65`. However, `devp2p` supports running multiple versions of the same wire protocol side-by-side, so rolling out `eth/65` does not require client coordination, since non-updated clients can keep using `eth/64`.
+
+This EIP does not change the consensus engine, thus does *not* require a hard fork.
+
+## Implementation
+<!--The implementations must be completed before any EIP is given status "Final", but it need not be completed before the EIP is accepted. While there is merit to the approach of reaching consensus on the specification and rationale before writing code, the principle of "rough consensus and running code" is still useful when it comes to resolving many discussions of API details.-->
+
+TODO
+
+## Security Considerations
+<!--All EIPs must contain a section that discusses the security implications/considerations relevant to the proposed change. Include information that might be important for security discussions, surfaces risks and can be used throughout the life cycle of the proposal. E.g. include security-relevant design decisions, concerns, important discussions, implementation-specific guidance and pitfalls, an outline of threats and risks and how they are being addressed. EIP submissions missing the "Security Considerations" section will be rejected. An EIP cannot proceed to status "Final" without a Security Considerations discussion deemed sufficient by the reviewers.-->
+
+None?
```

Comment:
Actually it totally invalidates the encryption which is based on the fact that both parties calculate MACs and encryption based on the stream of messages.

Reply:
Do you mind elaborating? Is this about known-plaintext attacks? Also, if this is a real problem then I suspect the `les` protocol does have this problem already since it already uses request ids.

Reply:
probably false alert since the order on the socket will still be known to build egress / ingree mac updates